src/Common/MoistThermodynamics/MoistThermodynamics.jl

```diff
@@ -85,6 +86,29 @@ air_pressure(ts::ThermodynamicState) =
   air_pressure(air_temperature(ts), air_density(ts), PhasePartition(ts))
 air_pressure(ts::PhaseDry) = air_density(ts) * gas_constant_air(ts) * air_temperature(ts)
 
+"""
+    linearized_air_pressure(ρ, ρe_tot, Φ[, q::PhasePartition])
+
+The air pressure, linearized around the reference state, from the equation of state
+(ideal gas law) where:
+
+ - `ρe_tot` total energy
+ - `Φ` gravitational potential
+ - `ρ` (moist-)air density
+and, optionally,
+ - `q` [`PhasePartition`](@ref). Without this argument the results are that of dry air.
+"""
+linearized_air_pressure(ρ::DT, ρe_tot::DT, Φ::DT, q::PhasePartition{DT}=PhasePartition{DT}(DT(0), DT(0), DT(0))) where {DT<:Real} =
+  ρ*DT(R_d)*DT(T_0) + DT(R_d)/DT(cv_d)*(ρe_tot - ρ*Φ - (ρ*q.tot - ρ*q.liq)*DT(e_int_v0) + ρ*q.ice*(DT(e_int_i0) + DT(e_int_v0)))
+
+linearized_air_pressure(ts::ThermodynamicState{DT}, ρe_tot::DT, Φ::DT) where {DT<:Real} =
+  linearized_air_pressure(air_density(ts), ρe_tot, Φ, PhasePartition(ts))
+
+function linearized_air_pressure(ts::PhaseDry{DT}, ρe_tot::DT, Φ::DT) where {DT<:Real}
+  ρ = air_density(ts)
```

Comment:
What is the rationale for treating density asymmetrically depending on whether the model is moist or dry? It is a state variable either way, so why is it not passed into this function in the same way as in the moist case?

Reply:
This `PhaseDry` function has a lower memory footprint because `PhasePartion` is not needed. Also, I pre-compute `ρ` here since it's used 2 times in the expression, the same function is called in either case (technically it's dispatched to the moist/dry case depending on the particular thermodynamic state).